nir/ir.py

```diff
@@ -261,6 +261,12 @@ def _forward_type_inference(self, debug=True):
                     k.replace('output', 'input'): v for k, v in pre_node.output_type.items()
                 }
 
+            # make sure that output nodes have output_type = input_type
+            if isinstance(post_node, Output):
+                post_node.output_type = {
```

Comment:
Hmmmm this looks like a hack

Reply:
it feels hacky, but the output node is the only node whose `input_type` field is irrelevant - so it's just a special case where we just have to set the `output_type` as well. a cleaner solution would be that Input and Output only have a `type` - but that would complicate many other things..